screens/auth/LoginScreen.js

```diff
@@ -80,102 +75,102 @@ export default class Login extends React.Component {
+    }
+    this.setState({ loginPermission });
+  };
+
   // This function will reformat the phone number to (XXX) XXX-XXXX and sign the user in if
   // the user is found.
-  handleSubmit() {
-    let formattedPhoneNumber = this.state.phoneNumber;
-    formattedPhoneNumber = formattedPhoneNumber.replace('[^0-9]', '');
-    formattedPhoneNumber = `(${formattedPhoneNumber.slice(
-      0,
-      3
-    )}) ${formattedPhoneNumber.slice(3, 6)}-${formattedPhoneNumber.slice(
-      6,
-      10
-    )}`;
-
-    lookupCustomer(formattedPhoneNumber, this.state.password)
-      .then(customerInfo => {
-        if (customerInfo) {
-          console.log(
-            'Customer lookup successful. Customer Record ID:',
-            customerInfo.custId
-          );
-          this.setState({ errorMsg: '' });
-          return customerInfo;
-        }
-        // If no records exist, resolves with null; set error message
-        this.setState({
-          errorMsg: 'Incorrect phone number or password. Please try again.',
-          phoneNumber: '',
-          password: '',
-        });
-        return null;
-      })
-      .then(customerInfo => {
-        if (customerInfo) {
-          updateCustomerPushTokens(customerInfo, this.state.token)
-            .then(customerId => {
-              if (customerId) {
-                this._asyncSignIn(customerId);
-              }
-              // Otherwise, lookup failed
-              return null;
-            })
-            .catch(err => console.error(err));
-        }
-        // Otherwise, lookup failed
-        return null;
-      })
-      .catch(err => console.error(err));
-  }
+  handleSubmit = async () => {
+    const { password, phoneNumber, token } = this.state;
 
-  onFocus(signUpField) {
-    const { indicators } = this.state;
-    indicators[signUpField] = fieldStateColors.FOCUSED;
-    this.setState({
-      indicators,
-    });
-  }
-
-  onBlur(signUpField) {
-    const { indicators } = this.state;
-    indicators[signUpField] = fieldStateColors.BLURRED;
-    this.setState({
-      indicators,
-    });
-  }
+    const formattedPhoneNumber = formatPhoneNumber(phoneNumber);
+    try {
+      let error = '';
+      const customers = await getAllCustomers(
+        `AND({Phone Number} = '${formattedPhoneNumber}', {Password} = '${password}')`
+      );
+      // Returns empty array if no customer is found
+      if (customers.length === 1) {
+        const customer = customers[0];
+        // If customer exists, we should update their push tokens
+        await updateCustomerPushTokens(customer, token);
+        // Log in
+        await this._asyncLogin(customer.id);
+      } else if (customers.length > 1) {
+        // In case of database malformation, may return more than one record
+        // TODO this message is a design edge case
+        error =
+          'Database error: more than one customer found with this login information. Please report an issue so we can fix it for you!';
```

Comment:
Well the 'report issue' link only shows in the hamburger menu which requires you to be logged in 😅 @12aschen